model/inline/src/main/scala/aqua/model/inline/TagInliner.scala

```diff
@@ -39,58 +39,65 @@ object TagInliner extends Logging {
    *
    * @param prefix Previous instructions
    */
-  enum TagInlined(prefix: Option[OpModel.Tree]) {
+  enum TagInlined[F](prefix: Option[OpModel.Tree]) {
 
     /**
      * Tag inlining emitted nothing
      */
-    case Empty(
+    case Empty[S](
       prefix: Option[OpModel.Tree] = None
-    ) extends TagInlined(prefix)
+    ) extends TagInlined[S](prefix)
 
     /**
      * Tag inlining emitted one parent model
      *
      * @param model Model which will wrap children
      */
-    case Single(
+    case Single[S](
       model: OpModel,
       prefix: Option[OpModel.Tree] = None
-    ) extends TagInlined(prefix)
+    ) extends TagInlined[S](prefix)
 
     /**
      * Tag inling emitted complex transformation
      *
      * @param toModel Function from children results to result of this tag
      */
-    case Mapping(
+    case Mapping[S](
       toModel: Chain[OpModel.Tree] => OpModel.Tree,
       prefix: Option[OpModel.Tree] = None
-    ) extends TagInlined(prefix)
+    ) extends TagInlined[S](prefix)
+
+    case After[S](process: () => State[S, TagInlined[S]], prefix: Option[OpModel.Tree] = None)
```

Comment:
Why return new `TagInlined`? This suggests that a tag could infinitely return new tag. Why not restricting return type to `OpModel.Tree`?

Reply:
Also what is the point in making `process` a function if it does not accept anything? Why not just make it of type `State[S, ...]`?